tests/unit/model-test.js

```diff
@@ -567,6 +569,106 @@ test('supports pushedData in root.deleted.uncommitted', function(assert) {
   });
 });
 
+test('supports canonical updates via pushedData in root.deleted.saved', function(assert) {
+  let { adapter } = env;
+
+  adapter.shouldBackgroundReloadRecord = () => false;
+  adapter.deleteRecord = () => {
+    return Ember.RSVP.resolve();
+  };
+
+  let record = run(() => store.push({
+    data: {
+      type: 'person',
+      id: '1',
+      attributes: {
+        isArchived: false
+      }
+    }
+  }));
+
+  run(() => {
+    record.destroyRecord().then(() => {
+      let currentState = record._internalModel.currentState;
+
+      assert.ok(currentState.stateName === 'root.deleted.saved',
+        'record is in a persisted deleted state');
+      assert.equal(get(record, 'isDeleted'), true);
+      assert.ok(store.peekRecord('person', '1') !== null, 'the deleted person is not removed from store (no unload called)');
+    });
+  });
+
+  run(() => {
+    try {
+      store.push({
+        data: {
+          type: 'person',
+          id: '1',
+          attributes: {
+            isArchived: true
+          }
+        }
+      });
+    } catch (e) {
+      assert.ok(false, e);
+    }
+
+    let currentState = record._internalModel.currentState;
+
+    assert.ok(currentState.stateName === 'root.deleted.saved',
+      'record is still in a persisted deleted state');
+    assert.ok(get(record, 'isDeleted') === true, 'The record is still deleted');
+    assert.ok(get(record, 'isArchived') === true, 'The record reflects the update to canonical state');
+  });
+});
+
+
+test('Does not support dirtying in root.deleted.saved', function(assert) {
+  let { adapter } = env;
+
+  adapter.shouldBackgroundReloadRecord = () => false;
+  adapter.deleteRecord = () => {
+    return Ember.RSVP.resolve();
+  };
+
+  let record = run(() => store.push({
+    data: {
+      type: 'person',
+      id: '1',
+      attributes: {
+        isArchived: false
+      }
+    }
+  }));
+
+  run(() => {
+    record.destroyRecord().then(() => {
+      let currentState = record._internalModel.currentState;
+
+      assert.ok(currentState.stateName === 'root.deleted.saved',
+        'record is in a persisted deleted state');
+      assert.equal(get(record, 'isDeleted'), true);
+      assert.ok(store.peekRecord('person', '1') !== null, 'the deleted person is not removed from store (no unload called)');
+    });
+  });
+
+  run(() => {
+    try {
+      set(record, 'isArchived', true);
+      assert.ok(false, 'Was unable to dirty a deleted record');
+    } catch (e) {
+      assert.ok(true, e.message);
```

Comment:
let's assert the specific error rather than merely that there was an error

Reply:
Will do